android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt

```diff
@@ -26,442 +26,486 @@ import com.amplitude.api.Plan
-    val eventObjectTypes: EventObjectTypesProperties? = null,
-    val eventWithAllProperties: EventWithAllPropertiesProperties? = null,
-    val eventWithArrayTypes: EventWithArrayTypesProperties? = null,
-    val eventWithConstTypes: EventWithConstTypesProperties? = null,
-    val eventWithDifferentCasingTypes: EventWithDifferentCasingTypesProperties? = null,
-    val eventWithEnumTypes: EventWithEnumTypesProperties? = null,
-    val eventWithOptionalArrayTypes: EventWithOptionalArrayTypesProperties? = null,
-    val eventWithOptionalProperties: EventWithOptionalPropertiesProperties? = null,
-    val group: GroupProperties? = null,
-    val identify: IdentifyProperties? = null
-)
-
-class ContextProperties()
-
-/**
- * Event to test schema validation
- */
-data class EventMaxIntForTestProperties (
-    /**
-     * property to test schema validation
-     */
-    val intMax10: Long
-)
-
-/**
- * Event w no properties description
- */
-class EventNoPropertiesProperties()
-
-/**
- * Event with Object and Object Array
- */
-data class EventObjectTypesProperties (
-    /**
-     * Property Object Type
-     */
-    val requiredObject: Any?,
-
-    /**
-     * Property Object Array Type
-     */
-    val requiredObjectArray: List<Any?>
-)
-
-/**
- * Event w all properties description
- */
-data class EventWithAllPropertiesProperties (
-    /**
-     * Event 2 Property - Optional String    *     * Examples:    * Some string, or another
-     */
-    val optionalString: String? = null,
-
+class Identify private constructor(
+    eventType: String,
+    eventProperties: Map<String, Any?>?,
+    options: EventOptions? = null
+) : Event(eventType, eventProperties, options) {
     /**
-     * Event 2 Property - Array
-     */
-    val requiredArray: List<String>,
-
-    /**
-     * Event 2 Property - Boolean
-     */
-    val requiredBoolean: Boolean,
-
-    /**
-     * Event 2 Property - Enum
-     */
-    val requiredEnum: EventWithAllPropertiesRequiredEnum,
-
-    /**
-     * Event 2 Property - Integer    *     * Examples:    * 5, 4, 3
-     */
-    val requiredInteger: Long,
-
-    /**
-     * Event 2 Property - Number
+     * Identify
+     *
+     * [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/0.0.0/Identify)
+     *
+     * Identify properties.
+     *
+     * @param requiredNumber Description for identify requiredNumber
+     * @param optionalArray Description for identify optionalArray
      */
-    val requiredNumber: Double,
+    constructor(
+        requiredNumber: Double,
+        optionalArray: Array<String>? = null
+    ) : this(
+        "Identify",
+        mapOf(
+            *(if (optionalArray != null) arrayOf("optionalArray" to optionalArray) else arrayOf()),
+            "requiredNumber" to requiredNumber
+        )
+    )
 
-    /**
-     * Event 2 Property - String
-     */
-    val requiredString: String
-)
+    fun options(userId: String? = null, deviceId: String? = null): Identify {
```

Comment:
We could also expose the `options(userId, ...)` method on the base Event class and then it would always be in sync as both those classes will be moved to the Amplitude-Android in the long run. Not sure if we could use generics like below, if not we could just return as a generic Event.

```
class EventOptions {
    val userId: String? = null,
    val deviceId: String? = null,
}

class Event {
  fun <E:Event>options(userId: String? = null, ....): E {
    return Event(this.eventType, this.eventProperties, EventOptions(...)) as E;
  }
}
```

Reply:
I did wanted to implement it in the same way but I do not like cast `Event(this.eventType, this.eventProperties, EventOptions(...)) as E` - it looks incorrect to cast base instance to child type.
Now I've fixed it in some other way.